redis/connection.py

```diff
@@ -1050,6 +1062,7 @@ def __init__(
         client_name=None,
         retry=None,
         redis_connect_func=None,
+        credential_provider: CredentialProvider = None,
```

Comment:
Same here: considering the default is `None`, I believe this should be
```suggestion
        credential_provider: Optional[CredentialProvider] = None,
```

Reply:
changed